drivers/network/tcpip/tcpip/fileobjs.c

```diff
@@ -472,29 +473,39 @@ NTSTATUS FileOpenAddress(
       if (Address->Address[0].Address[0].sin_port)
       {
           /* The client specified an explicit port so we force a bind to this */
-          AddrFile->Port = TCPAllocatePort(Address->Address[0].Address[0].sin_port);
+          nPort = TCPAllocatePort(Address->Address[0].Address[0].sin_port);
 
           /* Check for bind success */
-          if (AddrFile->Port == 0xffff)
+          if (nPort == (UINT)-1)
           {
+              AddrFile->Port = 0xffff;
```

Comment:
This is not needed, since AddrFile will be removed anyway

Reply:
Yes, right, my mistake (work fine, but useless)